refinery/tool_manager/models.py

```diff
@@ -135,3 +143,44 @@ class ToolDefinition(models.Model):
 
     def __str__(self):
         return "{}: {} {}".format(self.tool_type, self.name, self.uuid)
+
+
+@receiver(pre_delete, sender=ToolDefinition)
+def _tooldefinition_pre_delete(sender, instance, *args, **kwargs):
```

Comment:
Also, it's probably not a good idea to add the signal name to the handler name because signal name might change. It is better to describe what the function actually does ([example](https://github.com/refinery-platform/refinery-platform/blob/96a2cedb93ea6ab5c135efd3548f6e51ba8c4219/refinery/core/models.py#L136))